packages/itwin/grouping-mapping-widget/src/widget/components/hooks/useMappingsOperations.ts

```diff
@@ -30,22 +31,82 @@ const fetchMappings = async (
   }
 };
 
+const fetchExtractionStatus = async (
+  iModelId: string,
+  getAccessToken: GetAccessTokenFn,
+  extractionClient: IExtractionClient,
+  setIconStatus: React.Dispatch<React.SetStateAction<"negative" | "positive" | "warning">>,
+  setIconMessage: React.Dispatch<React.SetStateAction<string>>,
+  setExtractionMessageData: React.Dispatch<React.SetStateAction<ExtractionMessageData[]>>
+) => {
+  setIconStatus("warning");
+  setIconMessage("Extraction status pending.");
+  try {
+    const accessToken = await getAccessToken();
+    const extractions = await extractionClient.getExtractionHistory(accessToken, iModelId);
+    const jobId = extractions[0].jobId;
+    const status = await extractionClient.getExtractionStatus(accessToken, jobId);
+    if (status.containsIssues) {
+      setIconStatus("negative");
+      setIconMessage("Extraction contains issues. Click to view extraction logs.");
+      let logs = await extractionClient.getExtractionLogs(accessToken, jobId);
+      logs = logs.filter((log) => log.message != null);
+      const extractionMessageData = logs.map((log) =>
+        (
+          {
+            date: log.dateTime,
+            category: log.category,
+            level: log.level,
+            message: String(log.message),
+          }
+        ));
+      setExtractionMessageData(extractionMessageData);
+    } else {
+      setIconStatus("positive");
+      setIconMessage("Extraction Successful.");
+    }
+  } catch (error: any) { }
+};
+
 export interface MappingsOperationsProps extends GroupingMappingApiConfig {
   mappingClient: IMappingsClient;
 }
 
+export interface ExtractionMessageData {
+  date: string;
+  category: string;
+  level: string;
+  message: string;
+}
+
 export const useMappingsOperations = ({ iModelId, getAccessToken, mappingClient }: MappingsOperationsProps) => {
   const [showImportModal, setShowImportModal] = useState<boolean | undefined>(false);
   const [showDeleteModal, setShowDeleteModal] = useState<Mapping | undefined>(undefined);
   const [isLoading, setIsLoading] = useState<boolean>(true);
   const [mappings, setMappings] = useState<Mapping[]>([]);
+  const extractionClient = useExtractionClient();
   const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
   const [isTogglingExtraction, setIsTogglingExtraction] = useState<boolean>(false);
+  const [iconStatus, setIconStatus] = useState<"negative" | "positive" | "warning">("warning");
+  const [iconMessage, setIconMessage] = useState<string>("");
```

Comment:
I think `iconStatus` and `iconMessage` can be merged into one state.

Reply:
I have compressed these two in the object `extractionIconData`.